builder/frameworks/espidf.py

```
@@ -56,68 +56,55 @@
-    }
-
-    installed_packages = _get_installed_standard_pip_packages()
-    packages_to_install = []
-    for package, spec in deps.items():
-        if package not in installed_packages:
-            packages_to_install.append(package)
-        else:
-            version_spec = semantic_version.Spec(spec)
-            if not version_spec.match(installed_packages[package]):
-                packages_to_install.append(package)
+    installed_packages = _get_installed_pip_packages()
```

Ruff (0.11.9)

Undefined name `_get_installed_pip_packages`

(F821)